sdk/core/Azure.Core/src/RequestFailedException.cs

```diff
@@ -66,6 +67,19 @@ public RequestFailedException(int status, string message, string? errorCode, Exc
             ErrorCode = errorCode;
         }
 
+        internal RequestFailedException(int status, (string Message, string? ErrorCode) details):
```

Comment:
Do we have tests for this ctor?

Reply:
https://github.com/Azure/azure-sdk-for-net/issues/25554

Reply:
Let's make sure we have tests before it's time to ship.